src/Tools/VMC/EvtLibPXSec.cxx

```diff
@@ -0,0 +1,154 @@
+#include "Tools/VMC/EvtLibPXSec.h"
+#include "Tools/VMC/Utils.h"
+
+#include "Framework/Conventions/Units.h"
+
+#include "Framework/ParticleData/PDGLibrary.h"
+
+#include "TFile.h"
+#include "TGraph.h"
+
+using namespace genie;
+using namespace genie::vmc;
+
+//____________________________________________________________________________
+EvtLibPXSec::EvtLibPXSec() :
+XSecAlgorithmI("genie::vmc::EvtLibPXSec")
+{
+
+}
+//____________________________________________________________________________
+EvtLibPXSec::EvtLibPXSec(string config) :
+XSecAlgorithmI("genie::vmc::EvtLibPXSec", config)
+{
+
+}
+//____________________________________________________________________________
+EvtLibPXSec::~EvtLibPXSec()
+{
+
+}
+//____________________________________________________________________________
+double EvtLibPXSec::XSec(const Interaction* /*in*/,
+                         KinePhaseSpace_t /*kps*/) const
+{
+  std::cout << "EvtLibPXSec::XSec(): should not be reached" << std::endl;
+  abort();
+}
+
+//____________________________________________________________________________
+double EvtLibPXSec::Integral(const Interaction* in) const
+{
+  TGraph* g = GetXSec(in);
+  if(!g) return 0; // Reason already printed
+
+  const InitialState& init_state = in->InitState();
+  const double E  = init_state.ProbeE(kRfLab);
+  const Target& target = init_state.Tgt();
+
+  return g->Eval(E) * target.N() * 1e-38 * genie::units::cm2;
+}
+//____________________________________________________________________________
+bool EvtLibPXSec::ValidProcess(const Interaction* /*in*/) const
+{
+  return true;
+}
+//____________________________________________________________________________
+void EvtLibPXSec::Configure(const Registry & config)
+{
+  Algorithm::Configure(config);
+}
+//____________________________________________________________________________
+void EvtLibPXSec::Configure(string config)
+{
+  Algorithm::Configure(config);
+}
+
+//____________________________________________________________________________
+void EvtLibPXSec::LoadXSecs() const
+{
+  std::string libPath;
+  if(!GetParamDef( "LibraryPath", libPath, std::string() )){
+    LOG("ELI", pFATAL) << "Must specify 'LibraryPath'";
```

Comment:
Don't decide here how to kill the code. The code already has an exit if the parameter is not found. Just use GetParam( "LibraryPath", libPath ) ; and you'll be fine. 

See comments from Robert about the name of this variable

Reply:
Done